manual_flagging/initial_flagging.py

```diff
@@ -121,153 +140,373 @@
+
+for df_name, df in dfs_to_write.items():
+    print(f"{df_name}\n")
+    print(dfs_to_feature_creation[df_name]["df"].columns)
+
+    df_parameter = flg.get_parameter_df(
+        df_to_write=dfs_to_write[df_name]["df"],
+        df_ingest=dfs_to_feature_creation[df_name]["df"],
+        iso_forest_cols=dfs_to_write[df_name]["iso_forest_cols"],
+        stat_groups=dfs_to_write[df_name]["columns"],
+        dev_bounds=inputs["dev_bounds"],
+        ptax_sd=inputs["ptax_sd"],
+        rolling_window=inputs["rolling_window_months"],
+        date_floor=inputs["time_frame"]["start"],
+        short_term_thresh=flg_model.SHORT_TERM_OWNER_THRESHOLD,
+        min_group_thresh=inputs["min_groups_threshold"],
+        run_id=dfs_to_write[df_name]["run_id"],
+    )
+    # Standardize dtypes to prevent Athena errors
+    df_parameter = flg.modify_dtypes(df_parameter)
 
-# Standardize dtypes to prevent Athena errors
-df_parameters = flg.modify_dtypes(df_parameters)
+    dfs_to_write[df_name]["df_parameter"] = df_parameter
 
-flg.write_to_table(
-    df=df_parameters,
-    table_name="parameter",
-    s3_warehouse_bucket_path=os.getenv("AWS_S3_WAREHOUSE_BUCKET"),
-    run_id=run_id,
-)
+for df_name, df in dfs_to_write.items():
+    print(df_name)
 
-# Write to sale.group_mean table
-df_res_group_mean = flg.get_group_mean_df(
-    df=df_res_flagged, stat_groups=inputs["stat_groups"], run_id=run_id, condos=False
-)
+    df_group_mean = flg.get_group_mean_df(
+        df=dfs_flagged[df_name]["df"],
+        stat_groups=df["columns"],
+        run_id=df["run_id"],
+        condos=df["condos_boolean"],
+    )
 
-df_condo_group_mean = flg.get_group_mean_df(
-    df=df_condo_flagged, stat_groups=condo_stat_groups, run_id=run_id, condos=True
-)
+    dfs_to_write[df_name]["df_group_mean"] = df_group_mean
 
-df_group_mean_merged = pd.concat([df_res_group_mean, df_condo_group_mean]).reset_index(
-    drop=True
-)
+commit_sha = sp.getoutput("git rev-parse HEAD")
 
-flg.write_to_table(
-    df=df_group_mean_merged,
-    table_name="group_mean",
-    s3_warehouse_bucket_path=os.getenv("AWS_S3_WAREHOUSE_BUCKET"),
-    run_id=run_id,
-)
+for df_name, df in dfs_to_write.items():
+    print(df_name)
 
-# Write to sale.metadata table
-commit_sha = sp.getoutput("git rev-parse HEAD")
-df_metadata = flg.get_metadata_df(
-    run_id=run_id,
-    timestamp=timestamp,
-    run_type="initial_flagging",
-    commit_sha=commit_sha,
-)
+    # Write to sale.group_mean table
+    df_metadata = flg.get_metadata_df(
+        run_id=df["run_id"],
+        timestamp=df["timestamp"],
+        run_type="initial_flagging",
+        commit_sha=commit_sha,
+    )
 
-flg.write_to_table(
-    df=df_metadata,
-    table_name="metadata",
-    s3_warehouse_bucket_path=os.getenv("AWS_S3_WAREHOUSE_BUCKET"),
-    run_id=run_id,
-)
+    dfs_to_write[df_name]["df_metadata"] = df_metadata
+
+# - - - -
+# Write tables
+# - - - -
+
+for df_name, df in dfs_to_write.items():
```

Comment:
In the old structure, the order was as follows:
- create table
- write table
- create table
- write table

I think this new change is better:
- create all tables
- write all tables

When we had the out of memory error, it happened in-between writing different tables, which resulted in some meta data tables being written to prod, and some being lost forever, which meant that we had to manually delete the table and re-run. With this change, we create all of the table first, so if there happens to be an error in creating any of them, non of them will write, so we wouldn't have to manually nuke prod tables.